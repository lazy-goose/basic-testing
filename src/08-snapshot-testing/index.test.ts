import { generateLinkedList } from './index';

describe('generateLinkedList', () => {
  const generateFrom = ['item1', 'item2', 'item3'];
  const resultList = {
    value: 'item1',
    next: {
      value: 'item2',
      next: {
        value: 'item3',
        next: {
          value: null,
          next: null,
        },
      },
    },
  };

  test('should generate linked list from values 1', () => {
    expect(generateLinkedList(generateFrom)).toStrictEqual(resultList);
  });

  test('should generate linked list from values 2', () => {
    expect(generateLinkedList(generateFrom)).toMatchSnapshot();
  });
});
