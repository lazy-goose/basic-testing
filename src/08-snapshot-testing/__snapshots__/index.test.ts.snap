// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": null,
        "value": null,
      },
      "value": "item3",
    },
    "value": "item2",
  },
  "value": "item1",
}
`;
